### PR TITLE
Random `@Nullable` annotations: `Thread` and `AbstractPreferences`.

### PR DESCRIPTION
**Pull request by cpovirk**
I notice that we're inconsistent about whether to use `@Nullable` on the
`Runnable` parameter of `Thread`. I can see an argument that an
inconsistency is pragmatic from the perspective of a subclass that
overrides `run()`:

- There's no need for such a subclass to call call `super((Runnable)
  null)` when it could just call `super()`.
- There _is_ a need for it to pass a null `Runnable` to one of the
  "longer" constructors that provides a way to set things like
  `stackSize`, since there's no other way(?) to do that.

However, in this PR, I've added it universally.

(The Checker Framework would of course have a different perspective on
all of this.)

### CODE REVIEW

File: src/java.base/share/classes/java/lang/Thread.java [hunk 1]
```diff
@@ -764,7 +764,7 @@ public Thread(@Nullable ThreadGroup group, @Nullable Runnable target, String nam
      *
      * @since 9
      */
-    public Thread(ThreadGroup group, Runnable target, String name,
+    public Thread(@Nullable ThreadGroup group, @Nullable Runnable target, String name,
```
Comment (wmdietl) [inline src/java.base/share/classes/java/lang/Thread.java:767]: Looking through all the constructors:
- https://github.com/jspecify/jdk/blob/09f934f40c2c85afc11848dc68f130bac1bd659c/src/java.base/share/classes/java/lang/Thread.java#L497 could be `@Nullable`, but how useful is a Thread that does nothing?
- https://github.com/jspecify/jdk/blob/09f934f40c2c85afc11848dc68f130bac1bd659c/src/java.base/share/classes/java/lang/Thread.java#L585 could also be `@Nullable, but I'm not sure whether `If {@code null}, this thread's run method is invoked.` means anything different from `does nothing`. I guess subclasses of `Thread` could override `run` to do something different even with a null `Runnable`.
- https://github.com/jspecify/jdk/blob/09f934f40c2c85afc11848dc68f130bac1bd659c/src/java.base/share/classes/java/lang/Thread.java#L633 same as previous.

Should we add a comments to these or also make them `@Nullable`?

**Review by wmdietl — COMMENTED** [review]
> (The Checker Framework would of course have a different perspective on
all of this.)

I'm not so sure... the Javadoc for all these parameters say that null is a legal value and under no condition an NPE would be raised if null is passed.
Here, it's probably not so much about being conservative, but maybe "helpful": why would you create a `Thread` with a null `Runnable`? But on the helpfulness-side, JSpecify and CF don't need to vary much.

File: src/java.base/share/classes/java/lang/Thread.java (hunk 1 above)
Comment (cpovirk) [reply src/java.base/share/classes/java/lang/Thread.java:767]: Thanks for the nudge. I've had this thread starred in my mail for a while, but I keep not getting back to it.

I had indeed misremembered how `Thread.run()` worked, and apparently it never dawned on me to check my memory. I had been envisioning it as unconditionally calling `target.run()` where it in fact no-ops if `target` is `null`, as you say.

So I'm with you now this is just about helpfulness, not NPEs.

I suspect that the _behavior_ across constructors is meant to be presented as identical, even though the doc phrasing varies. We could still decide that a different annotation is more _helpful_ for one constructor than for another.

I don't actually know what the CF _or_ JSpecify position on helpfulness is :) My personal position has been that we have some design wiggle room except that we don't want for JSpecify annotations to call a type non-nullable when real users pass `null` in practice. That would probably suggest changing all of the `target` parameters to `@Nullable`, perhaps especially when the danger is "just" a silent no-op, not NPE.

(I'm assuming that _someone_ passes `null` to each of the constructors, even though I found little evidence of that just now, aside from one project whose tests create non-functional threads to pass to `Thread`-accepting debug methods. I guess that, if nothing else, the constructors call _one another_ with null arguments :))

I'd also add that anyone can always call the _public_ no-arg `Thread` constructor (which is meant for subclasses) and get the same behavior. So I might actually _prefer_ `new Thread(/* target= */ null)` if you actually really truly do want to do that (like for the tests I discussed above).

I've updated the PR to use `@Nullable Runnable target`. But I still don't have strong opinions here, so let me know if you've developed any :)

**Comment by cpovirk** [comment]
Thanks for checking on the CI. I've filed https://github.com/jspecify/jdk/issues/99.

**Pull request merged**